app/chip_tool/chip_tool.py

```diff
@@ -322,10 +322,12 @@ def __wait_for_server_exit(self) -> Optional[int]:
         return exit_code
 
     async def stop_chip_tool_server(self) -> None:
-        await self.__test_harness_runner.start()
-        await self.__test_harness_runner._client.send("quit()")
-        self.__wait_for_server_exit()
-        self.__server_started = False
+        if self.__server_started:
+            await self.start_runner()
+            await self.__test_harness_runner._client.send("quit()")
+            self.__wait_for_server_exit()
+            await self.stop_runner()
+            self.__server_started = False
```

Comment:
I'm wondering if we need some refactoring of this `ChipTool` class (maybe not needed for this PR)

But it's getting quite bloated, and it's really hard to follow what exactly is happening in a method like this.

I think we should split it in three classes:
- ChipCertBinsContainer
- ChipTool (covering running chip-tool/chip-app)
- MatterSDKRunner (for running yaml tests)

Reply:
What do you think @ccruzagralopes

Reply:
@ccruzagralopes can you make a new issue for this? An maybe propose some organization of the content in these classes, my suggestion above might not be the best one, so if you have other suggestions lets consider that too.

Reply:
Issue created: https://github.com/project-chip/certification-tool/issues/95

Reply:
Resolving this conversation because this is blocking the merge, is not related to changes meant to be done in this PR and Mikael is currently on holiday